src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -973,40 +975,40 @@ void TemplateTable::dastore() {
 
 // Pop 3 values from the stack and...
 void TemplateTable::aastore() {
+  tty->print_cr("aastore: %p", __ pc());
   transition(vtos, vtos);
 
   Label Lstore_ok, Lis_null, Ldone;
-  const Register Rindex    = R3_ARG1_PPC,
-                 Rarray    = R4_ARG2_PPC,
+  const Register Rindex    = R10_ARG0,
+                 Rarray    = R11_ARG1,
                  Rscratch  = R5_scratch1,
                  Rscratch2 = R6_scratch2,
-                 Rarray_klass = R5_ARG3_PPC,
+                 Rarray_klass = R12_ARG2,
                  Rarray_element_klass = Rarray_klass,
-                 Rvalue_klass = R6_ARG4_PPC,
-                 Rstore_addr = R31;    // Use register which survives VM call.
+                 Rvalue_klass = R13_ARG3,
+                 Rstore_addr = R14_ARG4;    // Use register which survives VM call.
 
-  __ ld_PPC(R25_tos, Interpreter::expr_offset_in_bytes(0), R23_esp); // Get value to store.
```

Comment:
They try to load value from `R23_esp`. But it should point to empty space.

Reply:
```
  static int expr_offset_in_bytes(int i) {
#if !defined(ZERO) && (defined(PPC) || defined(S390) || defined(SPARC))
    return stackElementSize * i + wordSize;  // both point to one word past TOS
#else
    return stackElementSize * i;
#endif
  }
```
-- it worked correctly for PPC. 
There are 8 uses of `expr_offset_in_bytes` in templateInterpter.

All of them should be fixed or `expr_offset_in_bytes` should consider RISCV same way as PPC. 
I think it's better to fix now -- it will fix some debugging time later